implementation/src/test/java/io/smallrye/config/ExpressionConfigSourceInterceptorTest.java

```diff
@@ -0,0 +1,57 @@
+package io.smallrye.config;
+
+import java.util.NoSuchElementException;
+
+import org.eclipse.microprofile.config.Config;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class ExpressionConfigSourceInterceptorTest {
+    @Test
+    public void simpleExpression() {
+        SmallRyeConfig config = (SmallRyeConfig) buildConfig("my.prop", "1234", "expression", "${my.prop}");
+
+        final String value = config.getValue("expression", String.class);
+        Assert.assertEquals("1234", value);
+    }
+
+    @Test
+    public void multipleExpressions() {
+        SmallRyeConfig config = (SmallRyeConfig) buildConfig("my.prop", "1234", "expression", "${my.prop}${my.prop}");
+
+        final String value = config.getValue("expression", String.class);
+        Assert.assertEquals("12341234", value);
+    }
+
+    @Test
+    public void composedExpressions() {
+        SmallRyeConfig config = (SmallRyeConfig) buildConfig("my.prop", "1234", "expression", "${${compose}}", "compose",
+                "my.prop");
+
+        final String value = config.getValue("expression", String.class);
+        Assert.assertEquals("1234", value);
+    }
+
+    @Test
+    public void defaultExpression() {
+        SmallRyeConfig config = (SmallRyeConfig) buildConfig("expression", "${my.prop:1234}");
```

Comment:
Could you add one more test where the default value is an expression e.g. `${prop1:${prop2}}`?

Reply:
Sure.